### DISCUSSION TITLE
Can't get bluetooth working with Qtile on Ubuntu 20.04.3 LTS

### DISCUSSION THREAD

**Discussion by JulienLecoq** [qtile/qtile#3484] [Q&A]
When I start my pc and connect to the qtile session, I can't get the bluetooth working.

Here is the output of **$HOME/.local/share/qtile/qtile.log**: 
`2022-04-08 02:01:05,687 ERROR libqtile loop.py:_handle_exception():L63 interface not found on this object: org.bluez.Device1`

Also, the settings app "can't find" Bluetooth:
![no_bluetooth_found](https://user-images.githubusercontent.com/54070985/162338854-b31a96b6-f922-4300-9177-439776783e00.png)

Bluetooth status:
![hciconfig](https://user-images.githubusercontent.com/54070985/162338851-76ed6cde-f2ab-4309-9c8a-9205cf279db6.png)
![service_bluetooth_status](https://user-images.githubusercontent.com/54070985/162338857-08234656-5906-4040-a98f-739ab6cf3ad1.png)
![bluetoothctl](https://user-images.githubusercontent.com/54070985/162339095-2426c6c0-fd30-4d83-b886-ad6a461d422b.png)
![rfkill](https://user-images.githubusercontent.com/54070985/162339179-dec50ac0-ec43-424e-b456-7483a97937ed.png)

Any idea of what could be the problem?
I have no idea of what are the meaning of the errors in the `bluetooth status` command.

**Comment by SweepingBishops** [comment]
In the bluetoothctl shell, what is the output of ```show```?

**Reply by JulienLecoq** [reply]
I have this:

![bluetoothctl_show](https://user-images.githubusercontent.com/54070985/162428614-a7564dad-6f42-4f8b-8e27-09bb6739c22c.png)

**Comment by elParaguayo** [comment]
How have you configured the widget? Have you got an `org.bluez` service on your system bus (use `d-feet` to check)?

**Reply by JulienLecoq** [reply]
I have used this configuration:
![bluetooth_widget](https://user-images.githubusercontent.com/54070985/162437858-bc528fc8-5a84-4f2a-87e4-18e0e010dd23.png)

The command: `sudo modprobe -r btusb && sudo modprobe btusb` makes the widget "off" text appears like this (next to the battery percentage):
![widget_icon](https://user-images.githubusercontent.com/54070985/162438251-e222227c-cb8a-45fd-8f4b-76a98dcacbea.png)
If I don't do this command, the widget "off" text doesn't  appears.

The `/dev_5C_5F_67_48_D1_12` does not appear in d-feet under `/org/bluez/hci0/` is that normal?
![d-feet](https://user-images.githubusercontent.com/54070985/162437871-ff70917a-4df8-4052-b844-c1284b51a1b1.png)

**Reply by elParaguayo** [reply]
OK - so there's obviously some driver issue which is why you need the modprobe command.

The `off` wording means that the controller is in a powered off state (but the widget is at least able to see it!).

As for your device not being listed in dbus, where did you get that device name from?

I notice that none of the devices listed in dbus are in your `bluetoothctl` screenshot...

**Reply by JulienLecoq** [reply]
You mean, where did I get `/dev_5C_5F_67_48_D1_12`? I get it from several commands, for instances `hciconfig -a` says that my Bluetooth address is: `5C:5F:67:48:D1:12`. So I just took this address and transformed it in the form `dev_XX_XX_XX_XX_XX_XX` as it is written there: https://docs.qtile.org/en/latest/manual/ref/widgets.html#bluetooth.

So I assume every devices listed in dbus must appear in `bluetoothctl show`?

It's very weird that Bluetooth is working as expected on the Ubuntu classic session but not on the qtile one. I will have to look more deeply to find what is causing this issue :/

**Reply by elParaguayo** [reply]
I wonder if there's a slight misunderstanding here. The device you put into the Bluetooth widget is the device you're trying to connect to your computer e.g. a pair of headphones, not the device of your computer's controller. 

When the headphones are connected, the name of the device will display in the widget.

It's just a widget to show the status of a particular device.

**Reply by JulienLecoq** [reply]
Ohh shit, yes I misunderstood. I thought it was the device of my computer's controller, to show the status of the Bluetooth on the computer, my bad.

**Reply by elParaguayo** [reply]
No problem. I think we could do a better job of clarifying this in the docs.

**Reply by JulienLecoq** [reply]
Yes :)
